control/metrics.go

```diff
@@ -250,6 +262,62 @@ func (m *metricType) Unit() string {
 	return m.unit
 }
 
+type catalogedPlugin struct {
```

Comment:
Is there really not another type that implements core.catalogedPlugin in control?